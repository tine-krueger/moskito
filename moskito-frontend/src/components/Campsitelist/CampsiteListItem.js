import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { ImPhone, ImEnvelop } from 'react-icons/im'
import FeatureListItem from './FeatureListItem'
import LikeButton from '../Button/LikeButton'
import { filterFeatures }from '../../services/featureService'


CampsiteListItem.propTypes = {
    campsite: PropTypes.object.isRequired
}

export default function CampsiteListItem({campsite}) {
    
    return (
        <CampItem>
            <h2>{campsite.name}</h2>
            <Address>
                <p>{campsite.street}, {campsite.postalCode} {campsite.place}</p>
                <Contact>
                    {campsite.telephone && <p><ImPhone/>{campsite.telephone}</p>}
                    {campsite.email && <p><ImEnvelop/> {campsite.email}</p>}
                    <a rel="noopener noreferrer" href={campsite.web} target='_blank'>{campsite.web}</a>
                </Contact>
            </Address>
            <FeatureList>
                {filterFeatures(campsite).map(feature => <FeatureListItem key={feature.id} feature={feature}/>)}
            </FeatureList>
            <LikeButton isPinned={campsite.pinned} id={campsite.id}/>
        </CampItem>
    )

}

const CampItem = styled.div`
    box-shadow: 2px 4px 8px -4px rgba(0, 0, 0, 0.6), 
                -2px -2px 2px 0px rgba(255, 255, 255, 0.5);
    border-radius: 20px;
    margin: 2em 1em;
    padding: .5em 1em;
    position: relative;


    h2 {
        width: 75%;
        margin-left: .3em;
    }

`
const Address = styled.div `
    margin-left: .5em;
`

const Contact = styled.div`
    p {
        position: relative;
        padding-left: 2em;
    }

    svg {
        width: 20px;
        height: 20px;
        color: #476c85;
        position: absolute;
        top: 0;
        left: 0;
    }

    a {
        display: block;
        margin-top: 1em;
    }
`
const FeatureList = styled.ul` 
   list-style: none;
   padding-left: 0;
   position: relative;
   margin-top: 3em;

   &::before{
    content: " ";
    display: inline-block;
    position: absolute;
    width: 90%;
    height: 5px;
    left: 5%;
    top: -25px;
    transition: all .2s ease;
    border-radius: 5px;
    box-shadow: 0.5px 0.5px 2px 0px rgba(0,0,0,0.4), 
                -1.25px -1.25px 2px 0px rgba(255,255,255,0.5);
   }
`