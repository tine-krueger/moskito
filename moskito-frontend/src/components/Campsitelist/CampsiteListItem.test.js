import { render } from '@testing-library/react'
import 'jest-styled-components'
import CampsiteListItem from './CampsiteListItem'

const campsite = {
    id: 1,
    name: "Nice Campsite",
    street: "Campsitestreet 1",
    postalCode: "12345",
    place: "Campsite Place",
    telephone: "",
    email: "mail@campsiteplace.de",
    web:"",
    pinned: false,
    features: [
        {
            type: "wlan",
            id: 1
        }
    ]
}
describe('Campsite-list item', () => {
    it('renders correctly', () => {
        const { container } = render(
            <CampsiteListItem
                campsite={campsite}
            />
        )
        expect(container).toMatchSnapshot()
    })

    it('shown the correct campsite information', () => {
        const { getByText, queryByTestId } = render(
            <CampsiteListItem
                campsite={campsite}
            />
        )
        expect(getByText(campsite.name)).toBeInTheDocument()
        expect(getByText(`${campsite.street}, ${campsite.postalCode} ${campsite.place}` )).toBeInTheDocument()
        expect(getByText(campsite.email)).toBeInTheDocument()
        expect(queryByTestId('mail')).toBeInTheDocument()
        expect(queryByTestId('phone')).not.toBeInTheDocument()
    })
})
