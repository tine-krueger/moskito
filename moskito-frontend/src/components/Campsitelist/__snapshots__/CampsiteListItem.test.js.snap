// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campsite-list item renders correctly 1`] = `
.c4 {
  position: absolute;
  right: 1em;
  top: 1em;
  height: 2.5em;
  width: 2.5em;
  border-radius: 10px;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 svg {
  width: 1.7em;
  height: 1.7em;
  stroke: #c97f63;
  stroke-width: 3em;
  fill: #c2d6d3;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 3px;
}

.c0 {
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  border-radius: 20px;
  margin: 2em 1em;
  padding: .5em 1em;
  position: relative;
}

.c0 h2 {
  width: 75%;
  margin-left: .3em;
}

.c1 {
  margin-left: .5em;
}

.c2 p {
  position: relative;
  padding-left: 2em;
}

.c2 svg {
  width: 20px;
  height: 20px;
  color: #476c85;
  position: absolute;
  top: 0;
  left: 0;
}

.c2 a {
  display: block;
  margin-top: 1em;
}

.c3 {
  list-style: none;
  padding-left: 0;
  position: relative;
  margin-top: 3em;
}

.c3::before {
  content: " ";
  display: inline-block;
  position: absolute;
  width: 90%;
  height: 5px;
  left: 5%;
  top: -25px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border-radius: 5px;
  box-shadow: 0.5px 0.5px 2px 0px rgba(0,0,0,0.4),-1.25px -1.25px 2px 0px rgba(255,255,255,0.5);
}

<div>
  <div
    class="c0"
  >
    <h2>
      Nice Campsite
    </h2>
    <div
      class="c1"
    >
      <p>
        Campsitestreet 1
        , 
        12345
         
        Campsite Place
      </p>
      <div
        class="c2"
      >
        
        <p>
          <svg
            data-testid="mail"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.5 2h-13c-0.825 0-1.5 0.675-1.5 1.5v10c0 0.825 0.675 1.5 1.5 1.5h13c0.825 0 1.5-0.675 1.5-1.5v-10c0-0.825-0.675-1.5-1.5-1.5zM6.23 8.6l-4.23 3.295v-7.838l4.23 4.543zM2.756 4h10.488l-5.244 3.938-5.244-3.938zM6.395 8.777l1.605 1.723 1.605-1.723 3.29 4.223h-9.79l3.29-4.223zM9.77 8.6l4.23-4.543v7.838l-4.23-3.295z"
            />
          </svg>
           
          mail@campsiteplace.de
        </p>
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        />
      </div>
    </div>
    <ul
      class="c3"
    />
    <div
      class="c4"
      data-testid="like-button"
    >
      <svg
        data-testid="pin"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M878.3 392.1L631.9 145.7c-6.5-6.5-15-9.7-23.5-9.7s-17 3.2-23.5 9.7L423.8 306.9c-12.2-1.4-24.5-2-36.8-2-73.2 0-146.4 24.1-206.5 72.3-15.4 12.3-16.6 35.4-2.7 49.4l181.7 181.7-215.4 215.2a15.8 15.8 0 0 0-4.6 9.8l-3.4 37.2c-.9 9.4 6.6 17.4 15.9 17.4.5 0 1 0 1.5-.1l37.2-3.4c3.7-.3 7.2-2 9.8-4.6l215.4-215.4 181.7 181.7c6.5 6.5 15 9.7 23.5 9.7 9.7 0 19.3-4.2 25.9-12.4 56.3-70.3 79.7-158.3 70.2-243.4l161.1-161.1c12.9-12.8 12.9-33.8 0-46.8z"
        />
      </svg>
    </div>
  </div>
</div>
`;
